tests/proofs/dexter/dexter-spec.md

```diff
@@ -122,6 +123,39 @@ If any of the conditions are not satisfied, the call fails.
       orBool IsUpdating
       orBool Sender =/=K CurrentManager
       orBool LQTAddress =/=K #Address("tz1Ke2h7sDdakHJQh8WX4Z372du1KChsksyU")
+      */
+```
+
+## Update Token Pool
+
+The contract queries its underlying token contract for its own token balance if the following conditions are satisfied:
+
+1.  the token pool is _not_ currently updating (i.e. `storage.selfIsUpdatingTokenPool = false`)
+2.  exactly 0 tez was transferred to this contract when it was invoked
+3.  the txn sender must be equal to the txn source
+4.  if we are running the FA2 version of Dexter, then check that the contract at address `storage.tokenAddress` has a well-typed FA2 `balance_of` entrypoint;
+    otherwise, check that the contract at address `storage.tokenAddress` has a well-typed FA12 `get_balance` entrypoint.
+
+```k
+  claim <k> #runProof(IsFA2, UpdateTokenPool) => . </k>
+        <stack> .Stack </stack>
+        <selfIsUpdatingTokenPool> false => true </selfIsUpdatingTokenPool>
+        <tokenAddress> TokenAddress:Address </tokenAddress>
+
+        <myaddr> SelfAddress </myaddr>
+        <myamount> #Mutez(Amount) </myamount>
+        <senderaddr> Sender </senderaddr>
+        <sourceaddr> Sender </sourceaddr>
+        <paramtype> #Type(nat) </paramtype>  // Why does this work????? Why would this be the type???
+        <knownaddrs> KnownAddresses </knownaddrs>
+        <returncode> 111 => ?_ </returncode> //  This is necessary if we hit an #Assume(false), I think that means we're on an "impossible" branch? So it's correct to include it?
```

Comment:
This should not be necessary, as far as I know. My understanding is: if we can reach a point where the returncode changed, we can reach a point where the return code has not changed and all other parts of the state are identical.

Reply:
well, this is to do with the fact that we hit an `#Assume(false)` it seems, and that only rewrites to `.` by rewriting the return-code, right?

Reply:
Oh, you're right. This rule:

```
rule <k> #Assume(false) ... </k>
     <returncode> 111 => 0 </returncode>
```

was added recently. This rule is no longer necessary; it should be deleted.